### PR TITLE
added concept of expose_mut. reference to protected

### PR DESCRIPTION
**Pull request by radumarias**
## Summary by CodeRabbit

- **New Features**
  - Introduced a new Continuous Integration (CI) workflow for automated build and testing processes.
  - Added platform-specific dependency management for Unix targets.

- **Bug Fixes**
  - Improved memory handling and error management for sensitive data in the `SecretBox` struct.

- **Chores**
  - Updated `.gitignore` to exclude IDE-specific files, enhancing version control hygiene.
  - Adjusted CI workflow to allow all jobs to run to completion, improving feedback on CI processes.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -25,29 +23,29 @@ pub struct SecretBox<S: Zeroize> {
...
 impl<S: Zeroize> Zeroize for SecretBox<S> {
     fn zeroize(&mut self) {
-        let len = mem::size_of_val(&*self.inner_secret);
+        self.inner_secret.as_mut().zeroize()
+    }
+}
+
+impl<S: Zeroize> Drop for SecretBox<S> {
+    fn drop(&mut self) {
+        let len = size_of_val(&*self.inner_secret);
```
Comment (Eyob94) [inline src/lib.rs:30]: Is this the same as `std::mem::size_of_val`? It's not compiling on my computer

File: src/lib.rs [hunk 2]
```diff
@@ -63,7 +61,7 @@ impl<S: Zeroize> From<Box<S>> for SecretBox<S> {
 impl<S: Zeroize> SecretBox<S> {
     /// Create a secret value using a pre-boxed value.
     pub fn new(boxed_secret: Box<S>) -> Self {
-        let len = mem::size_of_val(&*boxed_secret);
+        let len = size_of_val(&*boxed_secret);
```
Comment (Eyob94) [inline src/lib.rs:55]: Same here

File: src/lib.rs (hunk 1 above)
Comment (radumarias) [reply src/lib.rs:30]: should be as it use mem. but add full path or remove and use again

**Comment by Eyob94** [comment]
I changed it to `std::mem::size_of_val()` but the tests are failing

**Comment by radumarias** [comment]
test fails for me too, don't know why

**Pull request merged**